mindmeld/components/entity_resolver.py

```diff
@@ -1544,40 +1121,136 @@ class TfIdfSparseCosSimEntityResolver(EntityResolverBase):
+        self._aug_title_case = settings.get("augment_title_case", False)
+        self._aug_normalized = settings.get("augment_normalized", False)
+        self._aug_max_syn_embs = settings.get("augment_max_synonyms_embeddings", True)
+
+        self.ngram_length = 5  # max number of character ngrams to consider; 3 for elasticsearch
+        self._analyzer = kwargs.get("analyzer", None) or self._char_ngrams_plus_words_analyzer
+        self._vectorizer = TfidfVectorizer(analyzer=self._analyzer, lowercase=False)
         self._syn_tfidf_matrix = None
         self._unique_synonyms = []
 
-    def _char_ngram_and_word_analyzer(self, string):
-        results = self._char_ngram_analyzer(string)
-        # add words
+    def _char_ngrams_plus_words_analyzer(self, string):
+        """
+        Analyzer that accounts for character ngrams as well as individual words in the input
+        """
+        # get char ngrams
+        results = self._char_ngrams_analyzer(string)
+        # add individual words
         words = re.split(r'[\s{}]+'.format(re.escape(punctuation)), string.strip())
         results.extend(words)
         return results
 
-    def _char_ngram_analyzer(self, string):
+    def _char_ngrams_analyzer(self, string):
+        """
+        Analyzer that only accounts for character ngrams from size 1 to self.ngram_length
+        """
+        string = string.strip()
+        if len(string) == 1:
+            return [string]
+
         results = []
-        # give more importance to starting and ending characters of a word
-        string = f" {string.strip()} "
+        # give importance to starting and ending characters of a word
+        string = f" {string} "
         for n in range(self.ngram_length + 1):
             results.extend([''.join(gram) for gram in zip(*[string[i:] for i in range(n)])])
         results = list(set(results))
         results.remove(' ')
         # adding lowercased single characters might add more noise
         results = [r for r in results if not (len(r) == 1 and r.islower())]
+        # returns empty list of an empty string
+        return results
+
+    def find_similarity(self,
+                        src_texts,
+                        top_n=20,
+                        scores_normalizer=None,
+                        _return_as_dict=False,
+                        _no_sort=False):
+        """Computes sparse cosine similarity
+
+        Args:
+            src_texts (Union[str, list]): string or list of strings to obtain matching scores for.
+            top_n (int, optional): maximum number of results to populate. if None, equals length
+                of self._syn_tfidf_matrix
+            scores_normalizer (str, optional): normalizer type to normalize scores. Allowed values
+                are: "min_max_scaler", "standard_scaler"
+           _return_as_dict (bool, optional): if the results should be returned as a dictionary of
```

Comment:
Nit: Whats the standard you use to add the prefix `_` to the arg?

Reply:
I added it to differentiate it from other args in the sense that the args with underscore are not to be used generally except under special use-cases (underscore to emphasize their non-generic usage) and might possibly lead to unexpected outputs.